File: two_sum/Go/solution/ans_test.go
package solution_test

import (
	"sort"
	"testing"
	"twoSum/solution"
)

func TestSolution(t *testing.T) {
	inputs := []struct {
		nums   []int
		target int
		output []int
	}{
		{
			[]int{3, 4, 5, 6},
			7,
			[]int{0, 1},
		},
		{
			[]int{4, 5, 6},
			10,
			[]int{0, 2},
		},
		{
			[]int{5, 5},
			10,
			[]int{0, 1},
		},
	}

	for _, input := range inputs {
		ans := solution.TwoSum(input.nums, input.target)
		if !compareArrs(input.output, ans) {
			t.Fatalf(`
				Mismatching input. expected %v, got %v\n
				`, input.output, ans)
		}
	}
}

func compareArrs(expected []int, input []int) bool {
	if len(expected) != len(input) {
		return false
	}
	sort.Ints(expected)
	sort.Ints(input)

	for i := range len(expected) {
		if expected[i] != input[i] {
			return false
		}
	}
	return true
}

File: Questions/two_sum/Py/ans_test.py
import pytest
from ans import twoSum


def test_solution():
    inputs = [
        {
            "nums": [2, 7, 11, 15],
            "target": 9,
            "res": [0, 1],
        },
        {
            "nums": [3, 2, 4],
            "target": 6,
            "res": [1, 2],
        },
        {
            "nums": [3, 3],
            "target": 6,
            "res": [0, 1],
        }
    ]

    for input in inputs:
        assert twoSum(input["nums"], input['target']) == input['res']
